akka-http/src/main/scala/akka/http/scaladsl/unmarshalling/Unmarshaller.scala

```diff
@@ -154,7 +148,40 @@ object Unmarshaller
         s"Right failure: ${right.getMessage}, " +
         s"Left failure: ${left.getMessage}")
 
+  /**
+   * Signals that unmarshalling failed because the entity content-type did not match one of the supported ranges.
+   * This error cannot be thrown by custom code, you need to use the `forContentTypes` modifier on a base
+   * [[akka.http.scaladsl.unmarshalling.Unmarshaller]] instead.
+   */
+  final case class UnsupportedContentTypeException(
```

Comment:
Do we still want this to be a case class?

Reply:
Yes, as moving it to non-case class could cause more binary incompatibility issues

Reply:
We would indeed have to add some additional explicit methods (like `equals`), but it would avoid adding yet another automatically-generated `copy` that we'd have to handle when we change this type... OK though.